themes/src/main/resources/theme/base/login/messages/messages_en.properties

```diff
@@ -392,3 +392,8 @@ loggingOutImmediately=Logging you out immediately
 accountUnusable=Any subsequent use of the application will not be possible with this account
 userDeletedSuccessfully=User deleted successfully
 
+# Condition Errors in a Conditional flow
+conditionFailed=Preconditions were not satisfied.
```

Comment:
My vote is to simplify this and only use one failure message like "You are not allowed to authenticate" or "Access denied" or something like this. This is the only thing, which the typical user should know. He may not be interested (and sometimes he even should know) that the reason of the failed authentication is "required role" or "preconditions were not satisifed" or "required authenticators" . The typical user may not even know what these terms are :-) 

My vote would be to simplify this to only use single error message. This will hopefully simplify some logic in the authenticators itself around error messages?